Build and runtime fixes, remove packages folder, Parquet archiver updates and excel Signal Import feature

carterliukimleyhorn opened the pull request:
The packages folder being checked into Github was causing runtime issues for other developers.  We removed that folder and resolved several build issues, including missing references and mismatched .NET runtimes.

markKHA committed:
Merge pull request #3 from KimleyHorn/LTGAupdatesfor4_2_3

LTGA updates for 4.2.3

markKHA committed:
Revert "LTGA updates for 4.2.3"

markKHA committed:
Merge pull request #5 from KimleyHorn/revert-3-LTGAupdatesfor4_2_3

Revert "LTGA updates for 4.2.3"

markKHA committed:
Merge pull request #4 from udotdevelopment/master

Fix for Data Aggregation Memory Leak

markKHA committed:
Parquet support: new ParquetArchiver to facilitate database archival and support in MOE.Common to load Parquet files for reports. Deleting unused gitignore file (.gitignore is used instead).

markKHA committed:
Squashed commit of the following:

commit 686790b73f08a8f59a7b90e6506bce0f51dccfe6
Author: Andre Sanchez <asanchez@utah.gov>
Date:   Thu Jan 28 13:01:44 2021 -0700

    EOS files are deleted in the controller with ftps.

commit 21441ffcb29373dae7626f3cb451f302b9dd3d4e
Author: Derek Lowe <dlowe@utah.gov>
Date:   Sat Jan 23 15:03:08 2021 -0700

    Version 4.2.4

    Changes to Left Turn Gap Analysis, Menu seed, Chart order, Installer project, New aggregation

commit 72f5abb176386632eaa81e3639fb955ba8cf8c01
Author: Shane Johnson <31358034+shanejohnson-atspm@users.noreply.github.com>
Date:   Thu Nov 19 14:19:12 2020 -0700

    Installer (#90)

    * small bug fix in get maxtime records

    * Files and changes for an ATSPM installer

    The installer is created with Advanced Installer.
    Most of the config file changes are puctuation and white space.
    the csproj file changes are to makes sure each project builds to the  same platform (in this case, x86/release)

    Co-authored-by: udotdevelopment <dlowe@utah.gov>

commit e61c35528a6063023b8a487286fffeec5b36a128
Merge: e5f62a09 c92e52ea
Author: Derek Lowe <dlowe@utah.gov>
Date:   Thu Nov 19 14:01:56 2020 -0700

    Merge branch 'pr/78'

commit e5f62a098bdf1a30cabdb8ebd71c8f47e4f2496c
Merge: 77a0ea2f 78379728
Author: Derek Lowe <dlowe@utah.gov>
Date:   Thu Nov 19 13:58:55 2020 -0700

    Merge remote-tracking branch 'origin/master'

commit 78379728f9e39be2ec8163b9ad2d8b8d2bc81edf
Author: markKHA <mark.wood@kimley-horn.com>
Date:   Thu Nov 19 13:51:49 2020 -0700

    Update LTGA bin size (#80)

    * Build fixes. Note: All test projects have to be unloaded for VS to build solution.

    * LTGA options switched to cards for bootstrap update, change seed method for report to only work with detection types 4and 6.

    * Revert "LTGA updates for 4.2.3"

    * Update to bin size calcs

commit 77a0ea2f621364da489919652e2ae46a2ee0d2f2
Author: Derek Lowe <dlowe@utah.gov>
Date:   Thu Sep 10 11:32:16 2020 -0600

    fix for conflicts

commit c544c153066714abcd9dcda8c8f65d799e3251f2
Author: Andre Sanchez <asanchez@utah.gov>
Date:   Wed Jun 24 17:24:38 2020 -0600

    Changes in Aggregation

    This changes how the  Aggregation works to keep up with the daily data.  It is only doing the volume at this time.  Command line needs to be tested better.

markKHA committed:
Merge pull request #6 from udotdevelopment/master

Bring KH branch to current

markKHA committed:
Add FTP support for older Trafficware controllers, adding auto-retry function to Trafficware decoder, label fixes for Turning Movement Count report, adding 'Auto' Y-axis to Turning Movement Count report, updating AsyncGetMaxTimeRecords to .NET 4.7.2.

carterliukimleyhorn committed:
Merge branch 'master' of https://github.com/udotdevelopment/ATSPM into udotdevelopment-master

carterliukimleyhorn committed:
Merge branch 'udotdevelopment-master'

carterliukimleyhorn committed:
Build fixes.  Upgrade SPMTests and AsyncGetMaxTimeRecords to 4.7.2

carterliukimleyhorn committed:
Runtime bug fixes and remove packages folder

carterliukimleyhorn committed:
Merge branch 'trafficware_support_frisco' into Frisco

carterliukimleyhorn committed:
Merge pull request #12 from KimleyHorn/Frisco

Add FTP support for older Trafficware controllers, adding auto-retry …

carterliukimleyhorn committed:
Remove GetPinInfo

carterliukimleyhorn committed:
Merge branch 'feature_local_parquet' into kh_parquet

carterliukimleyhorn committed:
Revert "Merge branch 'feature_local_parquet' into kh_parquet"

This reverts commit b5d8792674a1663197d382e178796f0c292f1806, reversing
changes made to 0a0394fdc1b555ee5932c8b6706b81e28270c803.

carterliukimleyhorn committed:
Build fixes

carterliukimleyhorn committed:
Revert "Revert "Merge branch 'feature_local_parquet' into kh_parquet""

This reverts commit b534d9a4139ff5168754e8a0902e4ed0ba7ff3d6.

carterliukimleyhorn committed:
Merge pull request #16 from KimleyHorn/kh_parquet

Revert "Revert "Merge branch 'feature_local_parquet' into kh_parquet""

carterliukimleyhorn committed:
Add AWS, GCP, and Azure parquet support

carterliukimleyhorn committed:
merge conflicts

carterliukimleyhorn committed:
Merge fixes

carterliukimleyhorn committed:
Update AppConfig

markKHA committed:
Merge pull request #15 from KimleyHorn/feature_parquet_merging

Feature parquet merging

carterliukimleyhorn committed:
Add config files to gitignore, remove MOE.Common, MOEWcfServiceLibrary, ParquetArchiver config files from source control

markKHA committed:
Merge pull request #17 from KimleyHorn/feature_parquet_merging

Add config files to gitignore, remove MOE.Common, MOEWcfServiceLibrar…

markKHA committed:
Merge pull request #13 from KimleyHorn/trafficware_support_frisco

Trafficware support frisco

carterliukimleyhorn committed:
Fix bug when no AWS region specified, added case for off

carterliukimleyhorn committed:
Merge pull request #18 from KimleyHorn/feature_parquet_merging

Fix bug when no AWS region specified, added case for off (-1)

carterliukimleyhorn committed:
Parquet archiver  - Add foldername to config, target x64 to fix fatal error

carterliukimleyhorn committed:
Add Signal excel import functionality

carterliukimleyhorn committed:
Merge pull request #22 from KimleyHorn/SignalImport

Add Signal excel import functionality

markKHA commented:
@udotdevelopment We have enhanced this PR with new signal import functionality. Please let us know if you'd like to chat about this feature.

carterliukimleyhorn committed:
Merge pull request #19 from KimleyHorn/WaitTimeChartBugFix

Wait time chart bug fix

carterliukimleyhorn committed:
Merge pull request #21 from KimleyHorn/feature_parquet_merging

Parquet archiver - Add foldername to config, target x64 to fix fatal…

carterliukimleyhorn committed:
Merge pull request #23 from udotdevelopment/master

Bring udot changes

carterliukimleyhorn committed:
marge 4.1.3 and fix related bugs

carterliukimleyhorn committed:
Merge pull request #26 from KimleyHorn/4.1.3update

4.1.3 update

carterliukimleyhorn committed:
Add setting for file path prefix for parquet files.

carterliukimleyhorn committed:
Fix FTP collection for trafficware, change build output for MOE.Common

carterliukimleyhorn committed:
Implement watchdog exclusions

carterliukimleyhorn committed:
Add setting to hide IP from watchdog

carterliukimleyhorn committed:
Merge pull request #27 from KimleyHorn/ParquetArchiverDateSetting

Add setting for file path prefix for parquet files.

carterliukimleyhorn committed:
Merge pull request #29 from KimleyHorn/watchdog_manual_changes

Implement watchdog exclusions

carterliukimleyhorn committed:
Add SFTP from all controllers

carterliukimleyhorn committed:
Sort signals by primary name

carterliukimleyhorn committed:
ControllerEventLogsFromSignal can pull from parquet files

carterliukimleyhorn committed:
Merge pull request #32 from KimleyHorn/ApiUpdate

ControllerEventLogsFromSignal can pull from parquet files

markKHA committed:
Merge pull request #31 from KimleyHorn/SftpFromAllControllers

Add SFTP from all controllers

carterliukimleyhorn committed:
Fix incorrect labels shown on PCD chart

carterliukimleyhorn committed:
Merge pull request #34 from KimleyHorn/PCDfix

Fix incorrect labels shown on PCD chart

carterliukimleyhorn committed:
Add delete comment functionality, increment version to KH 4.3.2

carterliukimleyhorn committed:
Merge pull request #35 from KimleyHorn/ft_deletecomments

Add delete comment functionality, increment version to KH 4.3.2

carterliukimleyhorn committed:
Fix google storage archiver bugs

markKHA committed:
Merge pull request #37 from KimleyHorn/ArchiverBugFix

Fix google storage archiver bugs

carterliukimleyhorn committed:
Archiver now archives data for non-configured signals

carterliukimleyhorn committed:
Update development4.2.5.yml

carterliukimleyhorn committed:
Update development4.2.5.yml

carterliukimleyhorn committed:
Update development4.2.5.yml

carterliukimleyhorn committed:
Attempt to fix build pipeline

carterliukimleyhorn committed:
Merge branch 'master' of https://github.com/KimleyHorn/ATSPM

carterliukimleyhorn committed:
attempt to fix build pipeline

carterliukimleyhorn committed:
attempt to fix build pipeline

carterliukimleyhorn committed:
fix build pipeline

carterliukimleyhorn committed:
remove reference to spmapi that doesn't exist

carterliukimleyhorn committed:
Merge pull request #44 from KimleyHorn/ArchiverChanges

Archiver now archives data for non-configured signals

markKHA committed:
Bring in changes from gdot sandbox

markKHA committed:
config

carterliukimleyhorn committed:
Merge pull request #45 from KimleyHorn/SandboxChanges

Sandbox changes

carterliukimleyhorn committed:
add null check for approaches when saving signal

carterliukimleyhorn committed:
Merge pull request #46 from KimleyHorn/approachesFix

add null check for approaches when saving signal

carterliukimleyhorn committed:
Check previous day for most recent plan instead of previous hour

carterliukimleyhorn committed:
Increment version

carterliukimleyhorn committed:
Merge pull request #47 from KimleyHorn/reportPlanIssues

Check previous day for most recent plan instead of previous hour

carterliukimleyhorn committed:
Remove x2 labels on various charts, bug fix for generateadddata

carterliukimleyhorn committed:
remove labels from only TMC

carterliukimleyhorn committed:
remove x2 labels from arrivals on red and approach delay

Commit:
Updates to left turn gap analysis chart, patch notes

carterliukimleyhorn committed:
Merge pull request #49 from KimleyHorn/ltga_20231121

Updates to left turn gap analysis chart, patch notes

carterliukimleyhorn committed:
Merge pull request #48 from KimleyHorn/remove_x2_labels

Remove x2 labels from TMC, AoR, and AD charts, fix GenerateAddData bug

Commit:
Enhanced SFTP support.

carterliukimleyhorn committed:
Merge pull request #50 from KimleyHorn/sftp_datz_20231128

Enhanced SFTP support.

benpetru committed:
Add new PowerTools API Controller with endpoints for 3 reports that export data to a csv format
Split Monitor
Purdue Coordination Diagram
Purdue Split Failure

carterliukimleyhorn committed:
add d4 decoder and fix new signal default region

carterliukimleyhorn committed:
fix pipeline

carterliukimleyhorn committed:
Merge pull request #55 from KimleyHorn/d4decoder

add d4 decoder and fix new signal default region

carterliukimleyhorn committed:
Fix export page, update some required packages

carterliukimleyhorn committed:
Merge pull request #56 from KimleyHorn/export_fix

Fix export page, update some required packages

carterliukimleyhorn committed:
increment to 4.3.4, link pivot bug fix, update data access to pull from archive and db in same report.  generate add data validate coordinates. log raw data export to applicationevents

carterliukimleyhorn committed:
Merge pull request #58 from KimleyHorn/DataAccess

increment to 4.3.4, link pivot bug fix, update data access to pull fr…

carterliukimleyhorn committed:
log to db if signal has invalid coordinates

carterliukimleyhorn committed:
bulk import and fix bug when lane number is empty

carterliukimleyhorn committed:
Add export signal functionality

carterliukimleyhorn committed:
error fix

carterliukimleyhorn committed:
bug fixes related to export/import

carterliukimleyhorn committed:
Merge pull request #61 from KimleyHorn/bulk-import

bulk import/export and fix bug when lane number is empty

josephyoungkimleyhorn committed:
Adding basic error logging to AsyncGetMaxTimeRecords and code formatting

carterliukimleyhorn committed:
fix preempt bug

josephyoungkimleyhorn committed:
added the version change

carterliukimleyhorn committed:
Merge pull request #62 from KimleyHorn/AsyncGetMaxTimeRecordsErrorHandling

AsyncGetMaxTimeRecords Error Handling

carterliukimleyhorn committed:
Add ramp meter type and url

carterliukimleyhorn committed:
Merge pull request #54 from KimleyHorn/NewPowerTools

Add new PowerTools API Controller with endpoints for 3 reports

carterliukimleyhorn committed:
Merge pull request #63 from KimleyHorn/RampMeter

Add ramp meter type and url

carterliukimleyhorn committed:
remove extra atspmapi config

carterliukimleyhorn committed:
Fix newtonsoft bug

josephyoungkimleyhorn committed:
Added Console messaging for if a signal fails IP validation checks

josephyoungkimleyhorn committed:
Adding similar loogging for FTPFromAllControllers as in SFTPFromAllControllers

markKHA committed:
Remove logging - was causing SQL errors

josephyoungkimleyhorn committed:
Added Serilog Info and Error Logging to DecodeSiemensLogs. Prints daily log in logs folder.

josephyoungkimleyhorn committed:
Removed env changes

josephyoungkimleyhorn committed:
Removed more acidental settings changes

josephyoungkimleyhorn committed:
Last config change fix

josephyoungkimleyhorn committed:
All Gzip and SFTP ppk changes

josephyoungkimleyhorn committed:
Added Gzip decoding based on file extension

josephyoungkimleyhorn committed:
Added missing semi colon

josephyoungkimleyhorn committed:
Removing Gzip changes and adding missing semi-colon

josephyoungkimleyhorn committed:
Adding the new GetLatestVersionOfAllSignalsForSftp to InMemoryRepository

josephyoungkimleyhorn committed:
Added flag and Decoder

josephyoungkimleyhorn committed:
removing commented code

josephyoungkimleyhorn committed:
Deleted unused code

josephyoungkimleyhorn committed:
Cleaning up files after transfer

josephyoungkimleyhorn committed:
Merge pull request #67 from KimleyHorn/ASC3GzipDecoding

Added Gzip decoding based on file extension

josephyoungkimleyhorn committed:
Merge pull request #68 from KimleyHorn/SftpPpkAuthorization

Sftp ppk authorization and Controller type selection

josephyoungkimleyhorn committed:
Azure blob storage package update

josephyoungkimleyhorn committed:
skipCurrentLog functionality added

josephyoungkimleyhorn committed:
Added the DB Username from the controller type to the PPK authorizaiton instead of hardcoding it

josephyoungkimleyhorn committed:
Merge pull request #71 from KimleyHorn/SftpAddingUsernametoPPKAuth

SFTP PPK from DB

josephyoungkimleyhorn committed:
Giving Default IP, Lat Long values on import rather than erroring out

josephyoungkimleyhorn committed:
Merge pull request #72 from KimleyHorn/DefaultImportValuesIpLatLong

Default Lat Long IP

josephyoungkimleyhorn committed:
Updated Logging + Directory Navigation For Siemens Decoder

josephyoungkimleyhorn committed:
Updated signal auth to inclued the controller type UserName

carterliukimleyhorn committed:
Merge branch 'master' into SFTPFromAllControllersIPValidationLogging

carterliukimleyhorn committed:
Merge pull request #64 from KimleyHorn/SFTPFromAllControllersIPValidationLogging

Added Console messaging for if a signal fails IP validation checks

carterliukimleyhorn committed:
Merge pull request #65 from KimleyHorn/DecodeSiemensLogsLoggingAndYunexFiltering

Decode siemens logs logging and yunex filtering

carterliukimleyhorn committed:
Merge pull request #69 from KimleyHorn/AzureParquetArchiverPackageUpdate

Azure blob storage package update

carterliukimleyhorn committed:
Merge branch 'master' into SkipCurrentLog

carterliukimleyhorn committed:
Merge pull request #70 from KimleyHorn/SkipCurrentLog

skipCurrentLog functionality added

carterliukimleyhorn committed:
build fixed

carterliukimleyhorn committed:
SFTP bug fix

carterliukimleyhorn committed:
parquetarchiver - add serilog and azure support for missing days

carterliukimleyhorn committed:
parquetarchive - logging

josephyoungkimleyhorn committed:
(SFTP) added changes to sftp to allow any fingerprint, and added a vvariable gap between reading the folders and going through them after using the perfLogTranslate tool to decode

josephyoungkimleyhorn committed:
Merge Carters changes

josephyoungkimleyhorn committed:
Merge pull request #73 from KimleyHorn/SFTPFingerpringUpdate

Sftp fingerprint update

josephyoungkimleyhorn committed:
Sepac UTC to local time ignore and SFTP Delete files from remote origin logic added

josephyoungkimleyhorn committed:
Cleaned up comments and formatting

josephyoungkimleyhorn committed:
Reverting App.config

carterliukimleyhorn committed:
Merge pull request #74 from KimleyHorn/SepacDecoderTimezoneAndSftpPpkDeleteAdd

Sepac decoder timezone and sftp ppk delete add

carterliukimleyhorn committed:
parquet log to log file instead of txt

carterliukimleyhorn committed:
Merge branch 'master' of https://github.com/KimleyHorn/ATSPM

carterliukimleyhorn committed:
Migrate to tomtom map

carterliukimleyhorn committed:
guard invalid coordinates on map

carterliukimleyhorn committed:
light map

carterliukimleyhorn committed:
Merge pull request #75 from KimleyHorn/TomTom

Migrate to tomtom map

josephyoungkimleyhorn committed:
Map URL update to force base url rather than path url

josephyoungkimleyhorn committed:
Parquet Archiver directory and save retry logic + extra logging

carterliukimleyhorn committed:
Merge pull request #76 from KimleyHorn/ParquetArchiverRetry

Parquet Archiver directory and save retry logic + extra logging

josephyoungkimleyhorn committed:
Updated the Split Monitor to update the Yaxis max based on max split time rather than programmed split to accommodate Free Plans

carterliukimleyhorn committed:
fix duplicate signals on map, fix some javascript bugs

carterliukimleyhorn committed:
Merge pull request #77 from KimleyHorn/SplitMonitorYAxisMaxUpdate

Split Monitor YAxis update for free plans

carterliukimleyhorn committed:
resize map

carterliukimleyhorn committed:
remove timeout for resize on map load

carterliukimleyhorn committed:
remove signals url log

josephyoungkimleyhorn committed:
Corrected Signal Import IP address

josephyoungkimleyhorn committed:
removing check for is IP valid on FTP and SFTP

josephyoungkimleyhorn committed:
adding SignalFtpOptions Update

josephyoungkimleyhorn committed:
Adding sTfp -> sFtp csproj update and FluentFTP package upgrade

carterliukimleyhorn committed:
FTP and SFTP - Add setting to check IP

carterliukimleyhorn committed:
Add signals endpoint prefix setting

josephyoungkimleyhorn committed:
Checking in CsvReader with it's dependancies & changes

josephyoungkimleyhorn committed:
merging new changes to old branch

josephyoungkimleyhorn committed:
Updating git ignore and adding the d4 streamfile cleanout project

josephyoungkimleyhorn committed:
More Updates to try and get this thing to build

josephyoungkimleyhorn committed:
Moving the STFP -> SFTP

josephyoungkimleyhorn committed:
More updates

josephyoungkimleyhorn committed:
Git ignore updates

josephyoungkimleyhorn committed:
gitignore update and SFTP proj name update

josephyoungkimleyhorn committed:
Removing need for certificate

josephyoungkimleyhorn committed:
Adding missing interface reference

carterliukimleyhorn committed:
Merge pull request #80 from KimleyHorn/RemoveIPCheckOnFTP

SFTP name change, csv reader and d4 stream reader insert

carterliukimleyhorn committed:
Add serilog to CsvReader

carterliukimleyhorn committed:
add logging to csv reader to check for duplicates in file

josephyoungkimleyhorn committed:
Enhance CSV log processing: deduplication & error handling

Refactored CSV log workflow to improve duplicate detection by removing all but the first occurrence within files and logging removals. Added pre-insert database checks to prevent duplicate inserts, deleting files if duplicates are found. Improved logging clarity and robustness of file move/delete operations. Cleaned up code formatting and removed unused repository logic.

josephyoungkimleyhorn committed:
Adding missing fucntion

josephyoungkimleyhorn committed:
ATSPM New API Event code controller and formatting changes

josephyoungkimleyhorn committed:
Adding SPC files

josephyoungkimleyhorn committed:
Improve D4 FTP/SFTP handling, logging, and diagnostics

- Integrate Serilog for structured logging across projects
- Add/upgrade dependencies: FluentFTP, SSH.NET, Dapper, Serilog, etc.
- Auto-detect and persist FTP/SFTP connection type per signal
- Enhance FTP logic to resolve symlinks and nested directories
- Refactor file/directory operations for robustness and error handling
- Use Dapper for efficient DB access and ConnType updates
- Update repository interfaces and in-memory test stubs
- Miscellaneous code cleanup and event filtering adjustment

josephyoungkimleyhorn committed:
Update development4.2.5.yml

explicity telling the packages where to be built

josephyoungkimleyhorn committed:
Removing serilog target reference

josephyoungkimleyhorn committed:
Merge branch 'SCPFromD4Controllers' of https://github.com/KimleyHorn/ATSPM into SCPFromD4Controllers

josephyoungkimleyhorn committed:
Add SCPFromD4Controllers project and physical location support

- Added SCPFromD4Controllers project to solution and .sln configs
- ControllerType model: added PhysicalLocation property
- SignalFtpOptions: added UsePhysicalLocation flag and constructor param
- Program.cs: credentials now read from ControllerType, not file
- SQL query updated to include PhysicalLocation
- Refactored main loop to use Parallel.ForEach for all signals
- GetD4Files can override remote directory with PhysicalLocation
- Improved SFTP logic for password/PPK auth and error handling
- Refactored FTP/SFTP file transfer methods for clarity
- Removed large blocks of commented legacy code from Program.cs

carterliukimleyhorn committed:
Add new partition process md file

carterliukimleyhorn committed:
edit runbook

carterliukimleyhorn committed:
add columnstore index instructions

MckaylaGuzmanKH committed:
Add move option to NewAac3Decoder for archiving files instead of deleting

MckaylaGuzmanKH committed:
use Convert.ToBoolean for Move setting to match existing style

josephyoungkimleyhorn committed:
Merge pull request #84 from KimleyHorn/wisdot-decoder-move-option

Added Move feature to decoder for WisDot

carterliukimleyhorn committed:
Merge pull request #83 from KimleyHorn/SCPFromD4Controllers

Bunch of changes and the inclusion of the new SCPFromD4

josephyoungkimleyhorn committed:
Merge branch 'master' into CsvReaderDuplicateCheck

josephyoungkimleyhorn committed:
Added a fallback location for the different possible outcomes for a controller type

josephyoungkimleyhorn committed:
Checking in an update app config

carterliukimleyhorn committed:
Merge pull request #85 from KimleyHorn/ConfigPhysLocation

Added callback physical location

carterliukimleyhorn committed:
Merge pull request #81 from KimleyHorn/CsvReaderDuplicateCheck

Enhance CSV log processing: deduplication & error handling

josephyoungkimleyhorn committed:
Added ppk check on GetOrDetectConnType

josephyoungkimleyhorn committed:
Added changes to logging to include the db sink and reduced the minimu, severity level to error for db abd txt logging

josephyoungkimleyhorn committed:
add the regional controller type injection to the d4 controller type

josephyoungkimleyhorn committed:
Made small change to add brackets to avoid keywords on sql queries

carterliukimleyhorn committed:
add missing migration for controller type location.  fix map region filtering.  prevent migrations from re-running seed data after initial creation.

carterliukimleyhorn committed:
pipeline fix

carterliukimleyhorn committed:
Improve config, performance, and partitioning setup

Added `LogPath` and `LocalArchiveDirectory` keys to `Default.config` for enhanced logging and storage configuration.

Refactored `Program.cs` to optimize Google Cloud Storage checks by replacing client-side filtering with server-side prefix filtering, improving performance for large buckets.

Introduced `list-controller-event-log-partitions.sql` to list partitions and row counts for `dbo.Controller_Event_Log`, with detailed comments for debugging and monitoring.

Updated `partition-sliding-window-runbook.md` with instructions for fresh database deployments, ensuring partition-aligned table creation and compatibility with columnstore indexes.

Enhanced the pitfalls checklist in the runbook to emphasize best practices for partition alignment and index creation, reducing deployment errors.

josephyoungkimleyhorn committed:
Added Jursdiction path and parameter to db and SCPFromAllControllers Execution
